yourgov/components/FormMobileFoodVendorPermit/task1/FormTask1Step4.tsx

```diff
@@ -0,0 +1,105 @@
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { DatePicker } from '@ag.ds-next/react/date-picker';
+import { FormStack } from '@ag.ds-next/react/form-stack';
+import { Stack } from '@ag.ds-next/react/stack';
+import { TextInput } from '@ag.ds-next/react/text-input';
+import { DeepPartial } from '../../../lib/types';
+import { FormPageAlert } from '../FormPageAlert';
+import { hasMultipleErrors, parseDateField } from '../utils';
+import { StepActions } from '../StepActions';
+import { useGlobalForm } from '../GlobalFormProvider';
+import { useFormTask1Context } from './FormTask1Provider';
+import { FormTask1Container } from './FormTask1Container';
+import {
+	Task1Step4FormSchema,
+	task1Step4FormSchema,
+} from './FormTask1FormState';
+
+function transformDefaultValues(step?: DeepPartial<Task1Step4FormSchema>) {
+	const registrationExpiry = step?.registrationExpiry as
+		| string
+		| Date
+		| undefined;
```

Comment:
If you do that `DeepPartial` update mentioned earlier, this can just be:

```suggestion
	const registrationExpiry = step?.registrationExpiry;
```

Reply:
@ChrisLaneAU This one still required "as Date" so if you could look at that, that would be great